panic: invalid IP (length 0)

I updated my client this morning, and saw a panic come across the logs.

#### System information

Geth version: `v1.9.24-stable`
OS & Version: `linux-amd64/go1.15.5`
Commit hash : `cc05b050`

#### Backtrace

````
WARN [11-13|15:02:50.753] Synchronisation failed, dropping peer    peer=<REDACTED> err=timeout
panic: invalid IP (length 0)

goroutine 245853 [running]:
github.com/ethereum/go-ethereum/p2p/enode.nodeItemKey(0xe1e746b3728db12b, 0xf99e36e9d25005d, 0x7e0862ed8c5c18da, 0x3054f4210cdb23b2, 0x0, 0x0, 0x0, 0x132dc0a, 0x8, 0x30, ...)
	github.com/ethereum/go-ethereum/p2p/enode/nodedb.go:152 +0x49d
github.com/ethereum/go-ethereum/p2p/enode.(*DB).LastPingReceived(0xc01a7ca180, 0xe1e746b3728db12b, 0xf99e36e9d25005d, 0x7e0862ed8c5c18da, 0x3054f4210cdb23b2, 0x0, 0x0, 0x0, 0x1237f20, 0x1534101, ...)
	github.com/ethereum/go-ethereum/p2p/enode/nodedb.go:362 +0x78
github.com/ethereum/go-ethereum/p2p/discover.(*UDPv4).ensureBond(0xc015fb8000, 0xe1e746b3728db12b, 0xf99e36e9d25005d, 0x7e0862ed8c5c18da, 0x3054f4210cdb23b2, 0xc068b7bd70)
	github.com/ethereum/go-ethereum/p2p/discover/v4_udp.go:571 +0x6c
github.com/ethereum/go-ethereum/p2p/discover.(*UDPv4).RequestENR(0xc015fb8000, 0xc0660c56c0, 0x20300a, 0x20300a, 0x20300a)
	github.com/ethereum/go-ethereum/p2p/discover/v4_udp.go:342 +0x197
github.com/ethereum/go-ethereum/p2p/discover.(*UDPv4).Resolve(0xc015fb8000, 0xc0660c56c0, 0xaa17db)
	github.com/ethereum/go-ethereum/p2p/discover/v4_udp.go:178 +0x50
github.com/ethereum/go-ethereum/p2p.(*dialTask).resolve(0xc05e907800, 0xc0177e8000, 0x0)
	github.com/ethereum/go-ethereum/p2p/dial.go:519 +0x8d
github.com/ethereum/go-ethereum/p2p.(*dialTask).run(0xc05e907800, 0xc0177e8000)
	github.com/ethereum/go-ethereum/p2p/dial.go:484 +0xda
github.com/ethereum/go-ethereum/p2p.(*dialScheduler).startDial.func1(0xc05e907800, 0xc0177e8000)
	github.com/ethereum/go-ethereum/p2p/dial.go:459 +0x35
created by github.com/ethereum/go-ethereum/p2p.(*dialScheduler).startDial
	github.com/ethereum/go-ethereum/p2p/dial.go:458 +0x42a
INFO [11-13|15:02:51.856] Starting Geth on Ethereum mainnet... 
````

--- a/p2p/enode/nodedb.go
+++ b/p2p/enode/nodedb.go
@@ -61,6 +61,10 @@ const (
 	dbVersion        = 9
 )
 
+var (
+	errInvalidIP = errors.New("invalid IP")
+)
+
 var zeroIP = make(net.IP, 16)
 
 // DB is the node database, storing previously seen nodes and any collected metadata about
@@ -359,16 +363,25 @@ func (db *DB) expireNodes() {
 // LastPingReceived retrieves the time of the last ping packet received from
 // a remote node.
 func (db *DB) LastPingReceived(id ID, ip net.IP) time.Time {
+	if ip = ip.To16(); ip == nil {
+		return time.Time{}
+	}
 	return time.Unix(db.fetchInt64(nodeItemKey(id, ip, dbNodePing)), 0)
 }
 
 // UpdateLastPingReceived updates the last time we tried contacting a remote node.
 func (db *DB) UpdateLastPingReceived(id ID, ip net.IP, instance time.Time) error {
+	if ip = ip.To16(); ip == nil {
+		return errInvalidIP
+	}
 	return db.storeInt64(nodeItemKey(id, ip, dbNodePing), instance.Unix())
 }
 
 // LastPongReceived retrieves the time of the last successful pong from remote node.
 func (db *DB) LastPongReceived(id ID, ip net.IP) time.Time {
+	if ip = ip.To16(); ip == nil {
+		return time.Time{}
+	}
 	// Launch expirer
 	db.ensureExpirer()
 	return time.Unix(db.fetchInt64(nodeItemKey(id, ip, dbNodePong)), 0)
@@ -376,26 +389,41 @@ func (db *DB) LastPongReceived(id ID, ip net.IP) time.Time {
 
 // UpdateLastPongReceived updates the last pong time of a node.
 func (db *DB) UpdateLastPongReceived(id ID, ip net.IP, instance time.Time) error {
+	if ip = ip.To16(); ip == nil {
+		return errInvalidIP
+	}
 	return db.storeInt64(nodeItemKey(id, ip, dbNodePong), instance.Unix())
 }
 
 // FindFails retrieves the number of findnode failures since bonding.
 func (db *DB) FindFails(id ID, ip net.IP) int {
+	if ip = ip.To16(); ip == nil {
+		return 0
+	}
 	return int(db.fetchInt64(nodeItemKey(id, ip, dbNodeFindFails)))
 }
 
 // UpdateFindFails updates the number of findnode failures since bonding.
 func (db *DB) UpdateFindFails(id ID, ip net.IP, fails int) error {
+	if ip = ip.To16(); ip == nil {
+		return errInvalidIP
+	}
 	return db.storeInt64(nodeItemKey(id, ip, dbNodeFindFails), int64(fails))
 }
 
 // FindFailsV5 retrieves the discv5 findnode failure counter.
 func (db *DB) FindFailsV5(id ID, ip net.IP) int {
+	if ip = ip.To16(); ip == nil {
+		return 0
+	}
 	return int(db.fetchInt64(v5Key(id, ip, dbNodeFindFails)))
 }
 
 // UpdateFindFailsV5 stores the discv5 findnode failure counter.
 func (db *DB) UpdateFindFailsV5(id ID, ip net.IP, fails int) error {
+	if ip = ip.To16(); ip == nil {
+		return errInvalidIP
+	}
 	return db.storeInt64(v5Key(id, ip, dbNodeFindFails), int64(fails))
 }
 

